MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,129 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MediaBrowser.Common.Providers
+{
+    /// <summary>
+    /// Parsers for provider ids.
+    /// </summary>
+    public static class ProviderIdParsers
+    {
+        private const int ImdbMinNumbers = 7;
+        private const int ImdbMaxNumbers = 8;
+
+        /// <summary>
+        /// Parses an IMDb id from a string.
+        /// </summary>
+        /// <param name="text">The text to parse.</param>
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(string text, [NotNullWhen(true)] out string? imdbId)
```

Comment:
You can change all of the `string` parameters to be `ReadOnlySpan<char>` and then don't have to call the `AsSpan()` extension